cli/src/tests/node_test.rs

```diff
@@ -248,6 +248,24 @@ fn test_node_parent_of_child_by_field_name() {
     );
 }
 
+#[test]
+fn test_node_parent_with_inlined_parent() {
```

Comment:
I think this test could use a different name, since `inlined` isn't really part of the issue. Maybe `parent_with_zero_length_node`? I think the issue is that there are zero-length nodes inside of the `suite`,  right?

Reply:
yep you're right

Reply:
Yeah, I think the test should be called `test_parent_of_zero_width_node`. The `block` in this case (which is what you're testing the parent of) is a zero-width node, and that's why the problem was happening before.

The issue here doesn't have to do with inline subtrees. Inline subtrees are just a slightly more compact way of representing certain child-less subtrees.

I think your fix (avoiding descending into nodes without children) helps in this case (and most cases), but it's still possible that `ts_node_parent` could return the wrong value in the presence of zero-width nodes.

Fundamentally `ts_node_parent(node)` works by walking down the tree, starting from the root, trying to find the smallest possible node that contains the `node`'s byte range (and is not `node` itself). But if the node's byte range is empty, there are some edge cases.

I still think this fix improves the situation, and I'm glad you figured it out. I just think we should change the title of the test and 🚢 .

Reply:
Ahh gotcha, makes sense. Thanks for clarifying